### PR TITLE
Deprecate vendor reports

### PR DESCRIPTION
**Pull request by malteo**

### CODE REVIEW

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/VendorReportLineDTO.java [hunk 1]
```diff
@@ -6,6 +6,10 @@
 import java.math.BigDecimal;
 import java.util.Objects;
 
+/**
+ * @deprecated by {@link ResellingReportLineDTO}
+ */
+@Deprecated
 public class VendorReportLineDTO extends ReportLineDTO
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/VendorReportLineDTO.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/filters/VendorReportFilter.java [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 import java.util.Map;
 
+/**
+ * @deprecated by {@link ResellingReportFilter}
+ */
+@Deprecated
 public class VendorReportFilter extends ReportFilter
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/filters/VendorReportFilter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 2 issues
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/filters/VendorReportFilter.java [hunk 3]
```diff
@@ -4,6 +4,10 @@
 
 import java.util.Map;
 
+/**
+ * @deprecated by {@link ResellingReportFilter}
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/filters/VendorReportFilter.java]: rename in ProceedsReportFilter

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java [hunk 4]
```diff
@@ -315,6 +315,10 @@ public ResellingReportApi getResellingReportApi()
         return getApi( ResellingReportApi.class );
     }
 
+    /**
+     * @deprecated by {@link #getResellingReportApi()}
```
Comment (gionn) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java]: Rename in getProceedsReportApi

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java [hunk 5]
```diff
@@ -310,11 +310,24 @@ public ResellerPricingApi getResellerPricingApi()
...
+    public ProceedsReportApi getProceedsReportApi()
+    {
+        return getApi( ProceedsReportApi.class );
+    }
+
+    /**
+     * @deprecated by {@link #getProceedsReportApi()}
+     */
+    @Deprecated
     public ResellingReportApi getResellingReportApi()
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java [hunk 6]
```diff
@@ -310,11 +310,24 @@ public ResellerPricingApi getResellerPricingApi()
...
     {
         return getApi( ResellingReportApi.class );
     }
 
+    /**
+     * @deprecated by {@link #getProceedsReportApi()}
+     */
+    @Deprecated
     public VendorReportApi getVendorReportApi()
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/CloudesireClient.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/VendorReportApi.java [hunk 7]
```diff
@@ -12,6 +12,10 @@
 import java.util.List;
 import java.util.Map;
 
+/**
+ * @deprecated by {@link ResellingReportApi}
+ */
+@Deprecated
 public interface VendorReportApi
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/VendorReportApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/ResellingReportApi.java [hunk 8]
```diff
@@ -12,13 +12,17 @@
 import java.util.List;
 import java.util.Map;
 
+/**
+ * @deprecated by {@link ProceedsReportApi}
+ */
+@Deprecated
 public interface ResellingReportApi
```
Comment (AndreaVecchi) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/ResellingReportApi.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 7 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [ProceedsReportFilter.java#L49](https://github.com/ClouDesire/java-api-client/blob/3083006ddb2aeb65e8423c5162009c61f935da7c/cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/filters/ProceedsReportFilter.java#L49): Don't override a deprecated method or explicitly mark it as "@Deprecated". [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/ProceedsReportLineDTO.java [hunk 9]
```diff
@@ -31,6 +31,9 @@
     @ApiModelProperty( value = "The proceeds for the reseller", readOnly = true )
     private BigDecimal sellout;
 
+    @ApiModelProperty( value = "The proceeds for the vendor", readOnly = true )
+    private BigDecimal vendorIncome;
+
     @ApiModelProperty( value = "The costs for the distributor", readOnly = true )
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/ProceedsReportLineDTO.java]: wrong description?

**Pull request merged**